lib/Dialect/AIE/IR/AIEDialect.cpp

```diff
@@ -808,10 +815,10 @@ LogicalResult xilinx::AIE::BufferOp::verify() {
 
 // MemOp
 LogicalResult xilinx::AIE::MemOp::verify() {
+  Region &body = getBody();
   DenseSet<xilinx::AIE::DMAChannel> used_channels;
-
-  assert(getOperation()->getNumRegions() == 1 && "MemOp has zero region!");
-  assert(!getBody().empty() && "MemOp should have non-empty body");
+  if (body.empty())
+    return emitOpError("cannot have an empty body");
```

Comment:
Seems strange that this message differs from CoreOp... why not use the same language?

Reply:
The emitOpError() will actually start out by printing the Op at which the error occurred, in this case the CoreOp. 
Typically the syntax is:
CoreOp core;
core.emitOpError(...);
But inside the verify() we don't need to refer to the CoreOp.

Reply:
Most of the rest say "should have non-empty body".  Seems like consistency would be good here?

Reply:
Ah indeed! I will correct that. Thank you Steve!